services/brig/src/Brig/Provider/DB.hs

```diff
@@ -406,10 +406,16 @@ paginateServiceTags tags start size = liftClient $ do
     let start' = Name (maybe "" (toLower . fromName) start)
     let size'  = size + 1
     let tags'  = unpackTags tags
-    p <- queryAll start' size' tags'
+    p <- filterPrefix (fromName start') <$> queryAll start' size' tags'
     r <- mapConcurrently resolveRow (result p)
     return $! ServiceProfilePage (hasMore p) (catMaybes r)
   where
+    filterPrefix :: Text -> Page TagRow -> Page TagRow
+    filterPrefix prefix p = do
+        let prefixed = filter (\(Name n, _, _) -> prefix `isPrefixOf` (toLower n)) (result p)
+            more     = hasMore p && length prefixed == length (result p)
```

Comment:
why `length prefixed == length (result p)` ?

Reply:
The idea is that since we are now removing results (because they are lexicographically greater but not a prefix), the _prefixed_ number of results will always be `<=` than the ones returned by C*. So if the number of results with prefixes is smaller than the number of results that are lexicographically smaller, than there are no more possible results with that prefix.

Might be easier to show with some examples
```
let cassandraData q n = take n $ filter (>= q) $ sort ["e","d","c","b","aa","aad","aaae"]
let query p n = filter (isPrefixOf p) $ cassandraData p n
-- If we ask 5 results from cassandra, starting with "a", we will get 
λ> cassandraData  "a" 5
["aa","aaae","aad","b","c"]
λ> query "a" 5
["aa","aaae","aad"] -- in this case, isMore is True but there are no more results, so `length prefixed == length (result p)` would fail and we would return `has_more` as `False`
```

Perhaps a bit clearer would be `more = hasMore p && length prefixed > fromIntegral size`?

It's worth noting that, with this API, paging is not really possible, i.e., API clients cannot use the last result and paginate. They could either:
 * Request more results (in case `size=100` was not used)
 * Be more specific with the search query

Thus, I believe to still be useful to show that there are more results for the requested prefix

Reply:
ah, makes sense. Would it make sense to add another let binding with a comment for this, ala:
```Haskell
let ....
-- if they were all valid prefixes, there may be more in Cassandra
allValid = length prefixed == length (result p)
more = allValid && hasMore p
```
?

Reply:
> ah, makes sense. Would it make sense to add another let binding with a comment for this, ala:

👌